socs/common/moxa_serial.py

```diff
@@ -114,28 +114,33 @@ def readbuf_slow(self, n):
         try:
             for i in range(n):
                 msg += self.sock.recv(1)
-        except BaseException:
+        except (TimeoutError, BlockingIOError):
             pass
         self.sock.setblocking(1)  # belt and suspenders
         self.settimeout(self.__timeout)
         return msg
 
-    def readbuf(self, n):
+    def readbuf(self, n, max_loop=10):
         """Returns whatever is currently in the buffer. Suitable for large
         buffers.
 
         Args:
             n: Number of bytes to read.
 
         """
-        if n == 0:
-            return ''
-        try:
-            msg = self.sock.recv(n)
-        except BaseException:
-            msg = ''
-        n2 = min(n - len(msg), n / 2)
-        return msg + self.readbuf(n2)
+        n_current = n
+        msg = ''
+        for i in range(max_loop):
+            if n_current <= 0:
+                return msg
+            try:
+                msg_current = self.sock.recv(n)
+            except (TimeoutError, BlockingIOError):
+                msg_current = ''
+            msg += msg_current
```

Comment:
Looking back at the error that @ykyohei posted in https://github.com/simonsobs/socs/issues/891#issuecomment-2928017025, it looks like this will fail if `msg` actually has anything in it, as it will be a bytes object:
```
>>> msg = b'response from device'
>>> msg_current = ''
>>> msg += msg_current
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't concat str to bytes
```

Easiest fix is to make `msg_current = b''`.